operator/src/main/java/oracle/kubernetes/operator/helpers/ResourceVersion.java

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.helpers;
+
+public class ResourceVersion implements Comparable<ResourceVersion> {
+  private final String value;
+
+  private final String prefix;
+  private final Integer version;
+  private final String prerelease;
+  private final Integer prereleaseVersion;
+
+  public ResourceVersion(String value) {
+    this.value = value;
+
+    // version values must follow pattern of characters followed by digits
+    // if the initial character token is "v", then the value "alpha" or "beta"
+    // may follow the numeric token followed by another token of digits.  Anything
+    // else is illegal
+    if (value == null || value.length() < 1 || !Character.isAlphabetic(value.charAt(0))) {
+      throw new IllegalArgumentException();
+    }
+
+    boolean foundVersion = false;
+    int i = 1;
+    for (; i < value.length(); i++) {
+      if (Character.isDigit(value.charAt(i))) {
+        foundVersion = true;
+        break;
+      }
+    }
+
+    if (!foundVersion) {
+      throw new IllegalArgumentException("Version not found");
+    }
+
+    prefix = value.substring(0, i);
+
+    int n = i++;
+    for (; i < value.length(); i++) {
+      if (!Character.isDigit(value.charAt(i))) {
+        break;
+      }
+    }
+
+    version = Integer.valueOf(value.substring(n, i));
+
+    if (i < value.length()) {
+      String remainder = value.substring(i);
+      if (remainder.startsWith("alpha")) {
+        prerelease = "alpha";
+        i += 5;
+        n = i;
+        for (; i < value.length(); i++) {
```

Comment:
this is fine, but a regex might be nicer

Reply:
I have 97% code coverage, so I'm going to leave it alone. :)